### PR TITLE
Add query hook for any API method

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do?

When no command is found, Kourou will try to execute the given command with the `sdk:query` command.  

The first argument has to be the name of the controller and the action separated by a semicolon (eg `document:create`)

Kourou will try to infer common arguments like `index`, `collection`, `_id` or `body`.  

It will automatically infer and accept the following lists of arguments:
 - `<command> <index>`
    * _eg: `kourou collection:list iot`_
.

 - `<command> <index> <collection>`
    * _eg: `kourou collection:truncate iot sensors`_
.

 - `<command> <index> <collection> <body>`
    * _eg: `kourou bulk:import iot sensors '{bulkData: []}'`_
.

 - `<command> <index> <collection> <id>`
    * _eg: `kourou document:delete iot sensors sigfox-123`_
.

 - `<command> <index> <collection> <id> <body>`
    * _eg: `kourou document:create iot sensors sigfox-123 '{temperature: 42}'`_

Then any argument will be passed as-is to the `sdk:query` method.

### Other changes

  - add `--id` argument to `sdk:query`
 - remove `document:get`
 - remove `document:create`

### CODE REVIEW

File: src/hooks/command_not_found/apiMethod.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ *  - <index> <collection> <id> <body>
+ *  - <index> <collection> <body>
+ *  - <index> <collection> <id>
+ *  - <index> <collection>
+ *  - <index>
+ *
+ * This is mainly to match easily methods from the document, bulk, realtime
+ * and collection controllers.
+ *
+ * If one of the infered argument is passed as a flag, then Kourou will
```
Comment (Leodau) [inline src/hooks/command_not_found/apiMethod.ts:22]:
```suggestion
 * If one of the infered arguments is passed as a flag, then Kourou will
```

File: src/commands/collection/create.ts [hunk 2]
```diff
@@ -8,30 +8,27 @@ export default class CollectionCreate extends Kommand {
...
-      default: '{}'
-    }),
     ...kuzzleFlags
   }
 
   static args = [
     { name: 'index', description: 'Index name', required: true },
-    { name: 'collection', description: 'Collection name', required: true }
+    { name: 'collection', description: 'Collection name', required: true },
+    { name: 'body', description: 'ollection mappings and settings in JS or JSON format. Will be read from STDIN if available' },
```
Comment (Leodau) [inline src/commands/collection/create.ts:17]:
```suggestion
    { name: 'body', description: 'Collection mappings and settings in JS or JSON format. Will be read from STDIN if available' },
```

File: src/commands/collection/create.ts [hunk 3]
```diff
@@ -8,30 +8,27 @@ export default class CollectionCreate extends Kommand {
...
-      default: '{}'
-    }),
     ...kuzzleFlags
   }
 
   static args = [
     { name: 'index', description: 'Index name', required: true },
-    { name: 'collection', description: 'Collection name', required: true }
+    { name: 'collection', description: 'Collection name', required: true },
+    { name: 'body', description: 'Collection mappings and settings in JS or JSON format. Will be read from STDIN if available' },
```
Comment (alexandrebouthinon) [inline src/commands/collection/create.ts:17]:
```suggestion
    { name: 'body', description: 'Collection mappings and settings in JS or JSON format. Will be read from STDIN if available' }
```

Comment (Aschen) [reply src/commands/collection/create.ts:17]: Actually this is intended, it's so much easier when you end with a trailing comma to add new element or copy paste any element

**Pull request merged**